controllers/btpoperator_controller.go

```diff
@@ -1052,6 +746,32 @@ func (r *BtpOperatorReconciler) resourcesExist(ctx context.Context, namespaces *
 	return false, nil
 }
 
+func (r *BtpOperatorReconciler) cleanUpAllBtpOperatorResources(ctx context.Context) error {
+	gvks, err := ymlutils.GatherChartGvks(ChartPath)
+	if err != nil {
+		return err
+	}
+
+	if err := r.deleteAllInstalledResources(ctx, gvks); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) deleteAllInstalledResources(ctx context.Context, gvks []schema.GroupVersionKind) error {
+	for _, gvk := range gvks {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(gvk)
+		if err := r.DeleteAllOf(ctx, obj, client.InNamespace(ChartNamespace), managedByLabelFilter); err != nil {
+			if !k8serrors.IsNotFound(err) && !k8serrors.IsMethodNotSupported(err) && !meta.IsNoMatchError(err) {
```

Comment:
De Morgan? !(a || b || c) - which one is more readable?

Reply:
your suggestion is more readable, thanks